src/analysis/inner_product_space/adjoint.lean

```diff
@@ -0,0 +1,145 @@
+equivalence. -/
+def adjoint_aux : (E →L[𝕜] F) →L⋆[𝕜] (F →L[𝕜] E) :=
+(continuous_linear_map.compSL _ _ _ _ _ ((to_dual 𝕜 E).symm : normed_space.dual 𝕜 E →L⋆[𝕜] E)).comp
+  (to_sesq_form : (E →L[𝕜] F) →L[𝕜] F →L⋆[𝕜] normed_space.dual 𝕜 E)
+
+@[simp] lemma adjoint_aux_apply (A : E →L[𝕜] F) (x : F) :
+  adjoint_aux A x = ((to_dual 𝕜 E).symm : (normed_space.dual 𝕜 E) → E) ((to_sesq_form A) x) := rfl
+
+lemma adjoint_aux_inner_left (A : E →L[𝕜] F) (x : E) (y : F) : ⟪adjoint_aux A y, x⟫ = ⟪y, A x⟫ :=
+by { simp only [adjoint_aux_apply, to_dual_symm_apply, to_sesq_form_apply_coe, coe_comp',
+                innerSL_apply_coe]}
+
+lemma adjoint_aux_inner_right (A : E →L[𝕜] F) (x : E) (y : F) : ⟪x, adjoint_aux A y⟫ = ⟪A x, y⟫ :=
+by rw [←inner_conj_sym, adjoint_aux_inner_left, inner_conj_sym]
+
+variables [complete_space F]
+
+lemma adjoint_aux_adjoint_aux (A : E →L[𝕜] F) : adjoint_aux (adjoint_aux A) = A :=
+begin
+  ext v,
+  refine ext_inner_left 𝕜 (λ w, _),
+  rw [adjoint_aux_inner_right, adjoint_aux_inner_left],
+end
+
+@[simp] lemma adjoint_aux_norm (A : E →L[𝕜] F) : ∥adjoint_aux A∥ = ∥A∥ :=
+begin
+  refine le_antisymm _ _,
+  { refine continuous_linear_map.op_norm_le_bound _ (norm_nonneg _) (λ x, _),
+    rw [adjoint_aux_apply, linear_isometry_equiv.norm_map],
+    exact to_sesq_form_apply_norm_le },
+  { nth_rewrite_lhs 0 [←adjoint_aux_adjoint_aux A],
+    refine continuous_linear_map.op_norm_le_bound _ (norm_nonneg _) (λ x, _),
+    rw [adjoint_aux_apply, linear_isometry_equiv.norm_map],
+    exact to_sesq_form_apply_norm_le }
+end
+
+/-- The adjoint of a bounded operator from Hilbert space E to Hilbert space F. -/
+def adjoint : (E →L[𝕜] F) ≃ₗᵢ⋆[𝕜] (F →L[𝕜] E) :=
+linear_isometry_equiv.of_surjective
+{ norm_map' := adjoint_aux_norm,
+  ..adjoint_aux }
+(λ A, ⟨adjoint_aux A, adjoint_aux_adjoint_aux A⟩)
+
+local postfix `†`:1000 := adjoint
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_left (A : E →L[𝕜] F) (x : E) (y : F) : ⟪A† y, x⟫ = ⟪y, A x⟫ :=
+adjoint_aux_inner_left A x y
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_right (A : E →L[𝕜] F) (x : E) (y : F) : ⟪x, A† y⟫ = ⟪A x, y⟫ :=
+adjoint_aux_inner_right A x y
+
+/-- The adjoint is involutive -/
+@[simp] lemma adjoint_adjoint (A : E →L[𝕜] F) : A†† = A :=
+adjoint_aux_adjoint_aux A
+
+/-- The adjoint of the composition of two operators is the composition of the two adjoints
+in reverse order. -/
+@[simp] lemma adjoint_comp (A : F →L[𝕜] G) (B : E →L[𝕜] F) : (A ∘L B)† = B† ∘L A† :=
+begin
+  ext v,
+  refine ext_inner_left 𝕜 (λ w, _),
+  simp only [adjoint_inner_right, continuous_linear_map.coe_comp', function.comp_app],
+end
+
+/-- The adjoint is unique: a map `A` is the adjoint of `B` iff it satisfies `⟪A x, y⟫ = ⟪x, B y⟫`
+for all `x` and `y`. -/
+lemma eq_adjoint_iff (A : E →L[𝕜] F) (B : F →L[𝕜] E) :
+  (∀ x y, ⟪A x, y⟫ = ⟪x, B y⟫) ↔ A = B† :=
```

Comment:
If you don't have strong feelings, I think this `iff` would be more useful the other way around, so it can be used to `rw` goals of the form `A = B†`.

Reply:
Sure -- I was thinking of `apply` here, but you anyway have to use `mp` so we might as well switch it around.